lib/exposes.js

```diff
@@ -394,6 +394,14 @@ class Climate extends Base {
         return this;
     }
 
+    withHiveSystemMode(modes, access = a.ALL, description = 'Mode of this device') {
+        assert(!this.endpoint, 'Cannot add feature after adding endpoint');
+        const allowed = ['off', 'heat', 'auto', 'dry', 'fan_only', 'sleep', 'emergency_heating'];
+        modes.forEach((m) => assert(allowed.includes(m)));
+        this.features.push(new Enum('hive_system_mode', access, modes).withDescription(description));
```

Comment:
Why add a new `system_mode`? This won't even work since the property returned by `tz.hive_thermostat_system_mode` is `system_mode` and not `hive_system_mode`

Reply:
Hi,  I added a new `system_mode` as certain commands such as `heat` send additional data to the device. As for the property, it is incorrect (let me fix that!) but on my host, zigbee2mqtt still executes the new code despite the incorrect property key. Perhaps that is unintentional behaviour?

Thanks,

James

Reply:
Probably because there is also a `withSystemMode()` which matches with `hive_thermostat_system_mode`